Libraries/Image/RCTImageLoader.m

```diff
@@ -524,20 +528,25 @@ - (RCTImageLoaderCancellationBlock)loadImageWithURLRequest:(NSURLRequest *)image
                                           partialLoadBlock:(RCTImageLoaderPartialLoadBlock)partialLoadBlock
                                            completionBlock:(RCTImageLoaderCompletionBlock)completionBlock
 {
-    __block volatile uint32_t cancelled = 0;
+    __block atomic_bool cancelled = ATOMIC_VAR_INIT(NO);
+    // TODO: Protect this variable shared between threads.
     __block dispatch_block_t cancelLoad = nil;
     dispatch_block_t cancellationBlock = ^{
+        BOOL alreadyCancelled = atomic_fetch_or(&cancelled, 1);
+        if (alreadyCancelled) {
+            return;
+        }
         dispatch_block_t cancelLoadLocal = cancelLoad;
-        if (cancelLoadLocal && !cancelled) {
+        cancelLoad = nil;
```

Comment:
I added this line because it seems to have been missing from previous implementation. The pattern of "read cancel block into local variable, erase the __block variable, run the local variable" was copied from line 400 but it was missing this line, which defeats the purpose.